src/addons/__tests__/Grid.spec.js

```diff
@@ -231,6 +231,58 @@ describe('Grid', function() {
     });
   });
 
+  describe('Cell Navigation', function() {
+    describe('when cell selection looping is disabled', function() {
+      beforeEach(function() {
+        this.component = this.createComponent({enableCellSelectionLoop: false, enableCellSelect: true});
+      });
+
+      describe('when on last cell', function() {
+        beforeEach(function() {
+          this.component.setState({ selected: { idx: 3, rowIdx: 1 } });
+        });
+        it('selection should move to first cell in next row', function() {
+          this.simulateGridKeyDown('Tab');
+          expect(this.component.state.selected).toEqual({ idx: 0, rowIdx: 2 });
```

Comment:
Maybe put the logic for the new behaviours in a new method so that `moveSelectedCell`doesn't start to grow out of control? Something like:

```
if (isCellSelectionRowLoopEnabled || isCellSelectionGridLoopEnabled) {
   let {rowDelta, idx} = this.calculateNextPosition(...);
}
```